### ISSUE TITLE
pyright should give an error when assigning a type by mistake

### ISSUE THREAD

**Issue by shamrin** [microsoft/pyright#253]
**Describe the bug**
I made a typo in issue #250. I've accidentally used `=` instead of `:` in type annotation. I think pyright should have pointed out the problem.

**To Reproduce**
```python
from typing import Dict

d = Dict[str, int] = {}
d['one'] = 1
```

**Expected behavior**
I would expect pyright to give some error on line `d = Dict[str, int] = {}`

**VS Code extension or command-line**
Name: Pyright
Id: ms-pyright.pyright
Description: VS Code static type checking for Python
Version: 1.0.62
Publisher: ms-pyright
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-pyright.pyright

**Additional context**
Here's what mypy is saying about the problematic line:

```
/Users/shamrin/src/pyright/test.py:3: error: Too few arguments for "__setitem__" of "dict"
/Users/shamrin/src/pyright/test.py:3: error: Invalid index type "Tuple[Type[str], Type[int]]" for "dict"; expected type "Dict[Dict[<nothing>, <nothing>], <nothing>]"
```

I don't understand what it means, but at least it's some error message :)

**Comment by erictraut** [comment]
Thanks for the suggestion. 

I've added a check for assignments and deletions of generic types.

![Screen Shot 2019-09-18 at 8 24 58 AM](https://user-images.githubusercontent.com/7040122/65162266-d37a6080-d9ed-11e9-947f-2df4146f9de5.png)

Note that the assignment check applies only to those that use a subscript (index) operator. That's because non-subscripted types can be assigned without generating an exception — it's a legitimate operation (although probably not a good idea in most cases).

For example:
```
Dict = int
```

This will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.64, which I just published.
